cpu/cc2538/periph/i2c.c

```diff
@@ -159,7 +159,7 @@ static void _i2c_master_slave_addr(uint16_t addr, bool receive)
 {
     DEBUG("%s (%" PRIx16 ", %d)\n", __FUNCTION__, addr, (int)receive);
     assert(!(addr & 0x80));
-    I2CM_SA = (addr << 1) | receive;
+    I2CM_SA = (addr << 1) | (receive ? 0x1 : 0x0);
```

Comment:
receive is 0 or 1 anyways?

Reply:
`cppcheck` complains otherwise.
Maybe casting it to `(int)` would be a better solution?

Reply:
cppcheck was unhappy with an integer type and `bool` being in a bitwise operation. I could have explicitly casted the bool to the type of `I2CM_SA`. But translating it to an integer literal with the same value has the advantage, that this will continue to work when the type of `I2CM_SA` is every changed in the future.

I expect the compiler to generate the exact same machine code.

Reply:
> Maybe casting it to `(int)` would be a better solution?

`cpu/cc2538/include/cc2538.h:207`:
```C
#define I2CM_SA                     ( *(cc2538_reg_t*)0x40020000 ) /**< I2C Master Slave address */
```

`cpu/cc2538/include/cc2538.h:124`:
```C
typedef volatile uint32_t cc2538_reg_t; /**< Least-significant 32 bits of the IEEE address */
```

So casting to `(int)` would result in different sign. Casting to `cc2538_reg_t` would add the `volatile` qualifier. The use of `volatile` for the actual register `I2CM_SA` is correct, but for the parameter `receive` would result int he code to be less efficient. So if casting is preferred, it should be casted to `uint32_t`.

Reply:
well, if this makes cppcheck happy, I guess we can keep it. Probably there are even people who find it more readable. ;)